src/Components/List/List.jsx

```diff
@@ -2,21 +2,54 @@ import React from 'react';
 import { useHistory } from 'react-router-dom';
 import { db } from '../../lib/firebase.js';
 import { formatString } from '../../lib/helpers.js';
+import calculateEstimate from '../../lib/estimates';
 import dayjs from 'dayjs';
 import './List.css';
 
+const getNumberOfPurchases = (item) => {
+  if (item.numberOfPurchases === undefined) {
+    return 1;
+  } else {
+    return item.numberOfPurchases + 1;
+  }
+};
+
+const getLastPurchaseDate = (item, currentDate) => {
+  if (item.lastPurchased === null) {
+    return currentDate;
+  } else {
+    dayjs(item.lastPurchased.toDate());
+  }
+};
+
 export default function List({ items, token }) {
   let history = useHistory();
 
   const purchaseItem = (item) => {
     const normalizedName = formatString(item.name);
+
+    const lastInterval = (date1, date2) => {
+      return date1.diff(date2, 'h') / 24;
+    };
+
+    const numberOfPurchases = getNumberOfPurchases(item);
+    const currentDate = dayjs(new Date());
+    const lastPurchaseDate = getLastPurchaseDate(item, currentDate);
+
```

Comment:
Since you have both dates needed for lastInterval and the function is only used once, consider storing it as a variable instead.
```javascript
const lastInterval = currentDate.diff(lastPurchaseDate, 'h') / 24;
```

Reply:
Thanks for the suggestion Terry! We have refactored this function to a variable like you recommended.